### ISSUE TITLE
Quotes around variables on same line in templates are being parsed and eliminated

### ISSUE THREAD

**Issue by oucil** [ansible/ansible#76761]
### Summary

I have an SQL dump file which includes a number of variables for quick importing into a database, and the behaviour of ansible seems to have changed and has caused our template to stop working.  When more than one variable on the same line is surrounded by quotes, the inner quotes are eliminated but the outer quotes are left in place. 

For example, the following...

    INSERT INTO `my_table` VALUES (1, '{{ one }}', NULL, '{{ two }}');

...is now resulting in...

    INSERT INTO `my_table` VALUES (1, 'value-one, NULL, value-two');

Notice that the two single quotes in the middle have been removed. This breaks the SQL. As per the MySQL/ANSI standards, I'm supposed to be using single quotes for string literals, and backticks only for identifiers. I've seen similar behaviour in `json_query` strings and have been able to fix with backticks, however this is a template and I'm not sure why the quotes are being interpreted this way now.

### Issue Type

Bug Report

### Component Name

template

### Ansible Version

```console
ansible 2.9.27
  config file = ######/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.8 (default, Nov  9 2021, 14:44:26) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
```

### Configuration

```console
ANSIBLE_PIPELINING(######/ansible.cfg) = True
ANSIBLE_SSH_CONTROL_PATH(######/ansible.cfg) = /tmp/ansible-ssh-%%h-%%p-%%r
ANSIBLE_SSH_RETRIES(######/ansible.cfg) = 3
DEFAULT_HOST_LIST(######/ansible.cfg) = ['######/hosts']
DEFAULT_JINJA2_NATIVE(######/ansible.cfg) = True
DEFAULT_LOG_PATH(######/ansible.cfg) = /root/ansible.production.log
DEFAULT_REMOTE_USER(######/ansible.cfg) = root
DISPLAY_SKIPPED_HOSTS(######/ansible.cfg) = False
HOST_KEY_CHECKING(######/ansible.cfg) = False
INTERPRETER_PYTHON(######/ansible.cfg) = /usr/bin/python3.6/
RETRY_FILES_ENABLED(######/ansible.cfg) = False
RETRY_FILES_SAVE_PATH(######/ansible.cfg) = /root/.ansible-retry
```

### OS / Environment

NAME="Rocky Linux"
VERSION="8.5 (Green Obsidian)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.5"
PLATFORM_ID="platform:el8"

### Steps to Reproduce

```yaml
- tasks:
  - name: Copy SQL Dynamic Data Dump File for import
    template: src=data.sql.j2 dest=/tmp/data.sql
    vars:
      one: 'value-one'
      two: 'value-two'
```

### Expected Results

I expect to have ALL of the single quotes surrounding the variables, regardless of how many are on a line, to remain intact...

    INSERT INTO `my_table` VALUES (1, 'value-one', NULL, 'value-two');

### Actual Results

```console
Only the outermost quotes are maintained, and all inner quotes are removed...

    INSERT INTO `my_table` VALUES (1, 'value-one, NULL, value-two');
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by bcoca** [comment]
possible workaround:
```INSERT INTO `my_table` VALUES (1, {{"'" ~ one ~ "', NULL, '" ~ two ~ "'" }});```

**Comment by oucil** [comment]
> possible workaround: `` INSERT INTO `my_table` VALUES (1, {{"'" ~ one ~ "', NULL, '" ~ two ~ "'" }}); ``

The issue is that this is a huge file with hundreds of instances of lines with more than one variable on the same line, it would be hugely taxing to have to go back and rewrite this template with such a verbose solution if you know what I mean.

**Comment by sivel** [comment]
I believe this issue is a duplicate of https://github.com/ansible/ansible/issues/69242

But ultimately resolved in more recent versions of ansible-core by not using jinja2 native for the template module.

**Comment by sivel** [comment]
I'm a little foggy on how our concat plays with the jinja2 concat.  There was a change made in https://github.com/pallets/jinja/pull/1190 with how quotes were preserved.

However, in any case, since 2.9 is only receiving security fixes at this point, there will be no fixes made to 2.9 to rectify this behavior.

I recommend upgrading to ansible-core 2.11 or newer which shouldn't be affected by this, due to the fact that native is not used for the template module: https://github.com/ansible/ansible/pull/68560

**Comment by oucil** [comment]
@sivel Thanks for the added detail, I've just tried a run after commenting out the Jinja2 native setting and it seems to have completed without error.  It's only a small play within a MUCH larger playbook so I'll have to do some testing to make sure everything still works, but I appreciate the help identifying it and pointing out the change to native support.

**Comment by mkrizek** [comment]
> > I believe this issue is a duplicate of #69242
> > But ultimately resolved in more recent versions of ansible-core by not using jinja2 native for the template module.
> 
> Definitely looks to be the same, though I'm using Jinja 3.0.3, was the 2.11 a patch that would show up in i.e. 3.0.4?

It is fixed in Jinja2 version 2.11 and higher, including 3.0.3. Note that what matters is what Jinja2 version is installed on the controller, that is the machine from where you run ansible.

**Comment by mkrizek** [comment]
Closing as per above.

If you have further questions please stop by IRC or the mailing list:
* IRC: #ansible on irc.libera.chat
* mailing list: https://groups.google.com/forum/#!forum/ansible-project
